src/ros_vrpn_client.cpp

```diff
@@ -141,29 +163,31 @@ class Rigid_Body
 //  std::cout<<"tv.vel[0]"<<tv.vel[0]<<std::endl;
 //}
 
-//== Tracker Position/Orientation Callback ==--
-void VRPN_CALLBACK track_target(void *, const vrpn_TRACKERCB t)
+// Corrects measured target orientation and position for differing frame definitions.
+void inline correctForCoordinateSystem(const Eigen::Quaterniond& orientation_in,
+                                       const Eigen::Vector3d& position_in,
+                                       CoordinateSystem coordinate_system,
+                                       Eigen::Quaterniond* orientation_measured_B_W,
+                                       Eigen::Vector3d* position_measured_W)
 {
-  Eigen::Quaterniond qOrig(t.quat[3], t.quat[0], t.quat[1], t.quat[2]);
-  Eigen::Quaterniond qFix(0.70710678, 0.70710678, 0., 0.);
-
-  Eigen::Quaterniond orientation_measured_B_W;
-  Eigen::Vector3d position_measured_W;
+  // Rotation to correct between optitrack and vicon
+  const Eigen::Quaterniond kqOptitrackFix(0.70710678, 0.70710678, 0.0, 0.0);
```

Comment:
Nit: would really prefer this composed from an AngleAxisd around the correct axis of pi/2 instead of hardcoding the quaternion values. Makes it clearer what this does, and makes sure it's a properly normalized quaternion.
```
const Eigen::Quaterniond kqOptitrackFix(AngleAxisd(0.5*M_PI, Vector3f::UnitX()));
```
or whatevever is actually correct.